### PR TITLE
Deprecate clean=dist infavor of swift package reset

### PR DESCRIPTION
**Pull request by aciidb0mb3r**

### CODE REVIEW

File: Tests/CommandsTests/BuildToolTests.swift [hunk 1]
```diff
@@ -79,7 +79,8 @@ final class BuildToolTests: XCTestCase {
             }
 
             // Fully clean, and check for removal of both.
-            _ = try execute(["--clean=dist"], chdir: packageRoot)
+            let output = try execute(["--clean=dist"], chdir: packageRoot)
+            XCTAssert(output.contains("deprecated"))
```
Comment (ddunbar) [inline Tests/CommandsTests/BuildToolTests.swift]: We should also test `reset`, I suppose?

**Comment by aciidb0mb3r** [comment]
@swift-ci please test

**Comment by aciidb0mb3r** [comment]
@swift-ci please test linux

**Comment by aciidb0mb3r** [comment]
Unrelated linux failure

**Pull request merged**